Pipelines docs (#574)
kyma-project/api-gateway: docs/contributor/04-30-ci-cd.md

| [`Main Integration / Integration tests`]()                                                                                                                                            | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs Istio Module with Kyma CLI).                                    |
| [`Main Integration / Integration tests with Istio module`]()                                                                                                                          | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs Istio Module with the release manifest).                        |
| [`Main Integration / Upgrade tests (pull_request)`]()                                                                                                                                 | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version.   |

> | [`post-api-gateway-manager-custom-domain-integration-gcp`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-integration-tests.yaml#L6)  | Executes integration test suite which verify functional correctness of the API Gateway Manager while using custom domain on a Gardener cluster running on GCP. |

Reviewer: ```suggestion
| [`post-api-gateway-manager-custom-domain-integration-gcp`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-integration-tests.yaml#L6)  | Executes the integration test suite that verifies the functional correctness of Kyma API Gateway Operator. The job uses a custom domain and a Gardener GCP cluster. |
```

> | [`post-api-gateway-manager-custom-domain-integration-aws`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-integration-tests.yaml#L58) | Executes integration test suite which verify functional correctness of the API Gateway Manager while using custom domain on a Gardener cluster running on AWS. |

Reviewer: ```suggestion
| [`post-api-gateway-manager-custom-domain-integration-aws`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-integration-tests.yaml#L58) | Executes the integration test suite that verifies the functional correctness of the API Gateway Operator. The job uses a custom domain and a Gardener AWS cluster. |
```

| [`Main Integration / Slack Notification`]()                                                                                                                                           | Sends a Slack notification to the team's channel if any pipelines fail on the main branch.                                                                     |

## Pipelines running on the release branch

> The following CI jobs generate release artifacts and verify the functional correctness of API Gateway Manager.

Reviewer: ```suggestion
The following CI jobs generate release artifacts and verify the functional correctness of Kyma API Gateway Operator.
```

| [`rel-api-gateway-module-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L147)  | Builds API Gateway ModuleTemplate on the release and pushes it to the `prod` registry.      |

## Pipelines running on a schedule

The following CI jobs are scheduled to run at specific times.

| Name                                                         | Description                                                                                                                                                                                                                                         |
|--------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | [`Main Integration / Integration tests`]()                   | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs Istio Module with Kyma CLI). Runs periodically so it verifies correct behavior in case of new Istio Module release.                                  |

Reviewer: ```suggestion
| [`Main Integration / Integration tests`]()                   | Executes the main integration test suite for the API Gateway module on a k3d cluster. It installs the Istio module with Kyma CLI. The job runs periodically to verify that the API Gateway module behaves correctly with any new release of the Istio module.                                  |
```

> | [`Main Integration / Integration tests with Istio module`]() | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs Istio Module with the release manifest). Runs periodically so it verifies correct behavior in case of new Istio Module release.                      |

Reviewer: ```suggestion
| [`Main Integration / Integration tests with Istio module`]() | Executes the main integration test suite for the API Gateway module on a k3d cluster. It installs the Istio module with the release manifest. The job runs periodically to ensure that the API Gateway module behaves correctly with any new release of the Istio module.                      |
```

> | [`Main Integration / Upgrade tests (pull_request)`]()        | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version. Runs periodically so it verifies correct behavior in case of new Istio Module release. |

Reviewer: ```suggestion
| [`Main Integration / Upgrade tests (pull_request)`]()        | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version. The job runs periodically to ensure that the API Gateway module behaves correctly with any new release of the Istio module. |
```
